### ISSUE TITLE
BigIP provider crash when private_key is sensitive and known after apply

### ISSUE THREAD

**Issue by MasakariDOR** [F5Networks/terraform-provider-bigip#559]
### Environment

 * TMOS/Bigip Version:
 	BIG-IP 14.1.2.5 Build 0.0.3 Point Release 5
 * Terraform Version:
        Terraform 1.0.7
 * Terraform bigip provider Version:
        registry.terraform.io/f5networks/bigip v1.11.1

### Summary

We are provisioning virtual servers in full proxy with certificates created at runtime using Hashicorp Vault. The value of the private_key is marked as sensitive and is also "known after apply" and is rendered into the AS3 JSON using HCL. This produces the error :
```
│ Error: Provider produced inconsistent final plan
│
│ When expanding the plan for bigip_as3.application to include new values learned so far during apply, provider "registry.terraform.io/f5networks/bigip" produced an invalid new value for .as3_json:
│ was cty.StringVal(""), but now
│ cty.StringVal("{\"action\":\"deploy\",\"class\":\"AS3\" <SNIP...>
```
Once this error is encountered, the resources the bigip_as3 resource is dependent upon were already created and the sensitive value is now "known". Upon re-running, the sensitive value is handled correctly and the AS3 deployment is successful.

This bug is reproducible with or without Hashicorp Vault; we just need to ensure the value is "known" after apply.
We have a workaround to this issue but it is does block us from executing in a single run and requires 100% manual intervention in an otherwise streamlined automatic process. 

### Steps To Reproduce

1. Provide terraform resource config which you are facing trouble along with the output of it.
```
resource "bigip_as3"  "application" {
    provider = bigip
    as3_json = local.as3
}
```
2. To get to know more about the issue, provide terraform debug logs
[terraform.log](https://github.com/F5Networks/terraform-provider-bigip/files/7263237/terraform.log)

3. To capture debug logs, export TF_LOG variable with debug ( export TF_LOG= DEBUG ) before 
  runnning terraform apply/plan

4. As3/DO json along with the resource config( for AS3/DO resource issues )
```

terraform {
    required_providers {
        bigip = {
            source  = "f5networks/bigip"
            version = "1.11.1"
        }
    }
}

locals {
    as3 = format("%s%s%s","{\"action\":\"deploy\",\"class\":\"AS3\",\"declaration\":{\"class\":\"ADC\",\"id\":\"myApp1\",\"label\":\"myApp1\",\"remark\":\"HTTPS application\",\"schemaVersion\":\"3.20.0\",\"myApp1\":{\"App-App1\":{\"myApp1-443-CK\":{\"class\":\"Persist\",\"cookieName\":\"myApp1-CK\",\"duration\":180,\"persistenceMethod\":\"cookie\"},\"myApp1-443-CLIENTSSL\":{\"authenticationTrustCA\":{\"bigip\":\"/Common/ca-bundle.crt\"},\"certificates\":[{\"certificate\":\"myApp1.myCompany.com.au\"}],\"class\":\"TLS_Server\",\"label\":\"simplest decl requires just cert\"},\"myApp1-443-HM0\":{\"class\":\"Monitor\",\"interval\":5,\"monitorType\":\"https\",\"receive\":\"\",\"send\":\"GET / HTTP/1.0\\r\\n\",\"timeout\":16},\"myApp1-443-HM1\":{\"class\":\"Monitor\",\"monitorType\":\"tcp-half-open\",\"transparent\":false},\"myApp1-443-HTTP\":{\"class\":\"HTTP_Profile\",\"hstsInsert\":false,\"insertHeader\":{},\"trustXFF\":false,\"xForwardedFor\":true},\"myApp1-443-OC\":{\"class\":\"Multiplex_Profile\"},\"myApp1-443-POOL\":{\"class\":\"Pool\",\"loadBalancingMode\":\"round-robin\",\"members\":[],\"monitors\":[{\"use\":\"myApp1-443-HM0\"},{\"use\":\"myApp1-443-HM1\"}]},\"myApp1-443-SERVERSSL\":{\"class\":\"TLS_Client\",\"clientCertificate\":\"myApp1.myCompany.com.au\",\"label\":\"simplest decl requires just cert\",\"trustCA\":{\"bigip\":\"/Common/ca-bundle.crt\"}},\"myApp1-443-VS\":{\"class\":\"Service_HTTPS\",\"clientTLS\":\"myApp1-443-SERVERSSL\",\"label\":\"myApp1-443-VS\",\"persistenceMethods\":[{\"use\":\"myApp1-443-CK\"}],\"pool\":\"myApp1-443-POOL\",\"profileHTTP\":{\"use\":\"myApp1-443-HTTP\"},\"profileMultiplex\":{\"use\":\"myApp1-443-OC\"},\"profileTCP\":{\"bigip\":\"/Common/f5-tcp-lan\"},\"serverTLS\":\"myApp1-443-CLIENTSSL\",\"snat\":\"none\",\"virtualAddresses\":[\"10.100.0.28\"],\"virtualPort\":443},\"class\":\"Application\",\"myApp1.myCompany.com.au\":{\"certificate\":${var.certificate},\"chainCA\":\"-----BEGIN CERTIFICATE-----<SNIP>-----END CERTIFICATE-----\\n\",\"class\":\"Certificate\",\"privateKey\":",data.local_file.private_key.content,"},\"template\":\"generic\"},\"class\":\"Tenant\",\"defaultRouteDomain\":0}},\"persist\":true}")
}
provider "bigip" {
    address = "127.0.0.1"
}

resource "local_file" "private_key" {
    content = var.private_key
    filename = "./pkey.der"
}

data "local_file" "private_key" {
    filename = local_file.private_key.filename
}

variable "certificate" {
    default = "\"-----BEGIN CERTIFICATE-----<SNIP>-----END CERTIFICATE-----\\n\""
}
variable "private_key" {
    default = "\"-----BEGIN RSA PRIVATE KEY-----<SNIP>-----END RSA PRIVATE KEY-----\\n\""
    sensitive = true
}

resource "bigip_as3"  "application" {
    provider = bigip
    as3_json = local.as3
}
```

### Expected Behavior
When provisioning with a dynamic certificate/private_key, we expect to be able to render it into an AS3 and deploy it without a crash.

### Actual Behavior
A clear and concise description of what actually happens.
Please include any applicable error output.

When the private_key is both sensitive and "known after apply", the following errors:
```
│ Error: Provider produced inconsistent final plan
│
│ When expanding the plan for bigip_as3.application to include new values learned so far during apply, provider "registry.terraform.io/f5networks/bigip" produced an invalid new value for .as3_json:
│ was cty.StringVal(""), but now
│ cty.StringVal("{\"action\":\"deploy\",\"class\":\"AS3\" <SNIP...>
```

**Comment by trinaths** [comment]
Created INFRAANO-567 for internal tracking.

**Comment by MasakariDOR** [comment]
Any update on this? We have to run every deployment twice due to this error.

**Comment by RavinderReddyF5** [comment]
@MasakariDOR fix will be available in next release

**Comment by trinaths** [comment]
Created INFRAANO-662 for internal tracking. ETA ~ March 2022

**Comment by jskirde** [comment]
This is fixed in v1.13.0
